src/api/flaskr/service/learn/context_v2.py

```
@@ -197,6 +209,640 @@ def stream(
+        self.app.logger.info(
+            "preview document prompt | shifu_bid=%s | outline_bid=%s | prompt=%s",
+            shifu_bid,
+            outline_bid,
+            (document_prompt or "").strip(),
+        )
+        model, temperature = self._resolve_llm_settings(preview_request, outline, shifu)
+        document = preview_request.get_document() or (
+            outline.content if outline else ""
+        )
+        if not document:
+            raise ValueError("Markdown-Flow content is empty")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)